### ISSUE TITLE
$type attribute not generated for single objects using Newtonsoft.Json.TypeNameHandling.Auto; mode

### ISSUE THREAD

**Issue by AndreasGaus** [JamesNK/Newtonsoft.Json#2523]
I have configured $type attribute usage as "**Auto**": 

```
         services.AddMvc().AddNewtonsoftJson(options =>
         {
            options.SerializerSettings.ContractResolver = new DefaultContractResolver();
            options.SerializerSettings.TypeNameHandling = Newtonsoft.Json.TypeNameHandling.Auto; 
         });
```

Unfortunately, only inner objects are tagged automatically with the **$type** attribute as described in the [documentation](https://www.newtonsoft.com/json/help/html/SerializationSettings.htm#):

```
{
>>>> MISSING    "$type": "MyCompany.MyOuterClass, com.mycompany.myproject.rest", <<<<
    "DummyOuterStringProperty": "$type attribute above missing and added by hand!",
    "MyInnerObject": {
        "$type": "MyCompany.MyInnerClass, com.mycompany.myproject.rest",
        "DummyInnerStringProperty": "$type attribute above added automatically as described in Newtonsoft.Json documentation!"
    }
}
```

The outer object serialization of DefaultContractResolver does not add the **>>>>MISSING<<<<** $type attribute (this is added by hand in this example using a workaround).
If the workaround is done, both objects (outer and inner) are deserialized using the correct $type (polymorphism working now fine). Without the workaround (missing outer $type attribute), deserialization always creates a base class outer object and only the inner object is correctly subclassed.

**Question is, why is the $type attribute of the outer object not generated**. Both, outer and inner objects have sub classes and therefore the TypeNameHandling.Auto should also generate the outer $type if a subclassed object is to be serialized. One obvious difference is, that the outer class has no name, but only open / close brackets { }.

IMHO this seems to be a bug and I would be happy, if you can reproduce this and it could be fixed. Thanks in advance!

**Comment by TylerBrinkley** [comment]
Duplicate of https://github.com/JamesNK/Newtonsoft.Json/issues/990

**Comment by AndreasGaus** [comment]
Will this issue be solved in context of #990 ?

**Comment by BloggerMargaret** [comment]
Good day! I hope you all and your families are safe? The Covid-19 situation is really getting worst all over the globe, we all pray to almighty god to strength us, so that we can stay alive to testify, I brought good news that everyone can benefit from and also help the privilege, if you are a good kindest person, that also want to make the world a better place to live, now the freemasonry society are selecting the kindest people that are interested to join the society, and the good part is that freemasonry society they don't involved their self into evil things, they are not like the illuminati society, freemasonry society believe in god, they are grouped of people that come together to evaluate people and others original condition, once you become a member of the society they have sitting amount of benefit that we be given to you as the new member, and you should use 10% of the benefit to assist the privilege, and summit the credential to the lordship, I urge you if you have interest, find the official contact address below, you can be one of the lucky random, everyone has inside them a piece of good news, the good news is you don’t know how great you can be! How much you can love! What you can accomplish! And what your potential is, stay strong, stand up, have a voice.

WhatsApp; +31 687329133

Mail Address; info.masonic.oregon@gmail.com

Website Address; https://masonicoregon.wordpress.com
